packages/utils/telemetry-utils/src/perfTelemetryAggregator.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger, ITelemetryPerformanceEvent } from "@fluidframework/common-definitions";
+import { performance } from "@fluidframework/common-utils";
+
+class Measurements {
+    public executionCount: number = 0;
+    public totalDuration: number = 0;
+    public minDuration: number | undefined;
+    public maxDuration: number | undefined;
+
+    public reset() {
+        this.executionCount = 0;
+        this.totalDuration = 0;
+        this.minDuration = undefined;
+        this.maxDuration = undefined;
+    }
+}
+
+/**
+ * Maps a given identifier for a code block that is being subject to telemetry aggregation,
+ * to the runtime measurements for said code block.
+ */
+const statsMap: Map<symbol, Measurements> = new Map<symbol, Measurements>();
```

Comment:
If possible, we should not add globals / singletons.
Can this be done as a base class or mixin?

Reply:
And if there is only one thing per class to measure, then we do not need symbols at all

Reply:
Not sure I can visualize this functionality as a base class to be extended; as something that others could mix in, functionally yes, but my gut feeling doesn't like it? Taking that approach to the extreme I can imagine we end up with classes that mixin a bunch of different little pieces and it feels more complicated (and with no clear advantage that I can think of) than having a central component that takes care of this particular functionality.

I could turn the whole `perfTelemetryAggregator` into a class (in fact that was my first draft of it) and create a new instance in every "consumer" piece of code that wants to leverage it; a consumer class could keep one or more instances, to be able to measure however many separate things it needs, and with that approach symbols indeed become unnecessary. I went with a singleton because I thought it might be better for memory usage. If there are more transient objects/code-paths than persistent ones, which would be doing this kind of measurement, maybe instance-based would be better because most instances would eventually be garbage collected (by the JS runtime, not Fluid), but my impression is that we'll be using this mostly in objects that stay alive for the lifetime of the application? DDS would, right?

Reply:
your function should just return and object and it should be the responsibly of the consumer to track the measurement objects. t